### ISSUE TITLE
Pool's not finding any blocks - how to troubleshoot?

### ISSUE THREAD

**Issue by trololot** [dvandal/cryptonote-nodejs-pool#329]
Hi there. I've been running a couple of cryptonote-nodejs-pools on my home server, just solo-mining XTL and then ITNS. Got pointed 30 KH/s at it so was finding 1 block every 4-5 hours on average on XTL and every 3 hours on ITNS. Ran the XTL pool first for a couple of weeks and all was fine. Then tried running ITNS pool (separate machine, fresh install) and it was great for 3 days (was finding even more blocks than I was supposed to with my hashrate, about 10 blocks a day, with the average luck of 73%), but then it stopped finding any new blocks since yesterday's morning.

The network hashrate didn't change, all my rigs are working fine, I can't see anything wrong with the pool or coin daemon, no errors anywhere. The daemon is always at the same block height as any other pool/explorer out there, and the front-end also shows the correct height all the time. The VM I'm running the pool on is a clean Ubuntu 18.04 with 4 cores and 8 GB of ram dedicated to it, 1 miner with 8 workers. At first I thought it's just bad luck, but it's been 23 hours without a single block found, current effort is at almost 700% now. That doesn't seem right, it's like something's broken, but what? How does one troubleshoot issues like this - where to look?

**Comment by avense** [comment]
friend of luck changes lool
Mining is just the same.
There is no problem, in fact the crypto-coins are not made for mining alone, but as a team.

**Comment by trololot** [comment]
Yeah, thanks for the input, but I've been mining since 2011 and did quite a bit of solo mining back in the day so I know what bad luck is. :) This does not look like luck, this looks like something's broken in my setup, but I don't have enough experience with cryptonote-nodejs-pool to even know where to look when troubleshooting an issue like this. I've got enough hashrate to find 10 blocks a day, been finding blocks for both stellite and then intense just fine for 2 weeks with longest time before blocks for 8-9 hours (and that was only once), and then no block for 24+ hours? This is too much for bad luck - possible, but extremely unlikely. If the difficulty fluctuates a lot with something like nicehash hitting the coin's network regularly - then sure, something like this is totally possible, but that's not the case with ITNS - difficulty didn't change at all. I restarted the VM this night and found 2 blocks in 10 minutes, then left it running for the night. Just checked it right now, and there's been no block for 11 hours again. 

It looks like something gets stuck after a while. I did update redis via apt 3 days ago, but it was an update from the repository mentioned in the setup guide here (ppa:chris-lea/redis-server). That's pretty much the only thing that has changed after the initial pool setup. The main page here does have a warning about redis, but doesn't explain any specifics: 

"If you use old versions of Node.js or Redis that may come with your system package manager then you will have problems. Follow the linked instructions to get the last stable versions.
Redis warning: It'sa good idea to learn about and understand software that you are using - a good place to start with redis is data persistence."

What kind of "problems" exactly? Could there be something wrong with the latest redis from that chris-lea ppa? Anyone updated their ubuntu machines to this latest version (5:4.0.11-1chl1)? I've just checked my VM with the Stellite pool which I didn't update in a week or so (that one had no problems finding blocks on regular basis), and it has 5:4.0.10-1chl1 installed. I guess I'll try to revert my ITNS pool VM to this previous redis version. No, wait, just checked and it's no longer available in the repository. Installed 5:4.0.9-1 instead, although it's quite an old version from March 2018. But that's just me grasping at straws - the question is still the same: how to troubleshoot this kind of issues? Where to look?

**Comment by tetmyr** [comment]
ETN solo mining pool. I have 1 worker and two miners (ASIC), for the last week pool didn't find any block, ASIC "foundblock" indicator was showing they found from 1 to 3 blocks a day, but none of them went to the pool statistic.

**Comment by trololot** [comment]
So I rolled back the Redis update and went down from the latest 4.0.11 to 4.0.9 (it was 4.0.10 before, but I couldn't find that version, the repository maintainer had already deleted it) and so far the pool seems to be working fine. I don't know whether it's the latest Redis version that messed it up (maybe something's wrong with the build in chris lea ppa for ubuntu 18.04), or it was something else, or maybe even just super-super-bad luck, but still, if anyone will have this kind of issue, consider using another version of Redis (either 4.0.10/4.0.9, or maybe something newer than 4.0.11, whenever it shows up). 

![itns-pool-issues](https://user-images.githubusercontent.com/8095991/44002202-df5560c2-9e47-11e8-95de-2ed83d61ca27.png)

**Comment by pcca-matrix** [comment]
I have the same problem with my ITNS pool ! 
maybe it's itns related.

**Comment by trololot** [comment]
> I have the same problem with my ITNS pool !
maybe it's itns related.

What version of redis are you using?

**Comment by pcca-matrix** [comment]
@trololot 
redis 4.0.9

**Comment by DigitalDeviator** [comment]
If it's possible to find a block with 6% luck, it's also possible to have 600% bad luck. (or even more)

**Comment by pcca-matrix** [comment]
the problem is definitely the intense network not a pool issue ...
a lot of pool are closed for this reason
I will do the same , this coin is time and hash wasting
